### PR TITLE
Deneb container changes beacon-chain

### PR DESCRIPTION
**Pull request by enriavil1**
Adding container changes for Deneb: https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/beacon-chain.md

Changed to use pointers for BlobSideCar
 
 #7389

### CODE REVIEW

File: cmd/erigon-cl/core/transition/proccess_blob_side_car.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package transition
```
Comment (Giulio2002) [inline cmd/erigon-cl/core/transition/proccess_blob_side_car.go:1]: would not add it until cl specs are working, remove it for now

File: cl/cltypes/eth1_block.go [hunk 2]
```diff
@@ -29,9 +29,10 @@ type Eth1Block struct {
 	Extra         []byte
 	BaseFeePerGas [32]byte
 	// Extra fields
-	BlockHash    libcommon.Hash
-	Transactions [][]byte
-	Withdrawals  types.Withdrawals
+	BlockHash     libcommon.Hash
+	Transactions  [][]byte
+	Withdrawals   types.Withdrawals
+	ExcessDataGas *big.Int
```
Comment (Giulio2002) [inline cl/cltypes/eth1_block.go:35]: keep this as [32]byte as i did with baseFeePErGas, CL-wise it easier to mantain

File: cl/cltypes/eth1_block.go [hunk 3]
```diff
@@ -66,11 +67,15 @@ func NewEth1BlockFromHeaderAndBody(header *types.Header, body *types.RawBody) *E
 		BlockHash:     header.Hash(),
 		Transactions:  body.Transactions,
 		Withdrawals:   body.Withdrawals,
+		ExcessDataGas: header.ExcessDataGas,
 	}
-	if header.WithdrawalsHash == nil {
-		block.version = clparams.BellatrixVersion
-	} else {
+
+	if block.ExcessDataGas != nil {
```
Comment (Giulio2002) [inline cl/cltypes/eth1_block.go:73]: this is wrong you wanna set both excessDataGas and WithdrawalHash

File: cl/cltypes/beacon_blob_side_car.go [hunk 4]
```diff
@@ -79,14 +79,14 @@ func (b *BlobSideCar) EncodingSizeSSZ() int {
 	return 131_256
```
Comment (Giulio2002) [inline cl/cltypes/beacon_blob_side_car.go:79]: is it that big?

**Review by Giulio2002 — APPROVED** [review]
LGTM just answer the comment

File: cl/cltypes/beacon_blob_side_car.go (hunk 4 above)
Comment (enriavil1) [reply cl/cltypes/beacon_blob_side_car.go:79]: Yeah it is humongous

**Comment by Giulio2002** [comment]
lint fails too

File: cl/cltypes/beacon_blob_side_car.go (hunk 4 above)
Comment (enriavil1) [reply cl/cltypes/beacon_blob_side_car.go:79]: size of blob alone is 32*4096 = 131,072

**Comment by Giulio2002** [comment]
answered on discord

**Pull request merged**